### PR TITLE
[Documentation] Istio Module documentation clarification

### PR DESCRIPTION
**Pull request by cyrille-leclerc**
## What does this PR do?

Clarify docs. Co written with @ChrsMark .

## Why is it important?

Prevent misunderstandings

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: metricbeat/docs/modules/istio.asciidoc [hunk 1]
```diff
@@ -11,21 +11,20 @@ This file is generated! See scripts/mage/docs_collector.go
...
-For versions after `1.5`, `istiod` and `proxy` metricsets can be used.
+When using Istio `1.4-`, then the `mesh`, `mixer`, `pilot`, `galley`, `citadel` metricsets should be used. The Istio
+module collects metrics from these
+Istio https://istio.io/v1.4/docs/tasks/observability/metrics/querying-metrics/#about-the-prometheus-add-on[prometheus exporters endpoints].
 
-`istiod` collects metrics directly from Istio Daemon while `proxy` collects from each of the proxy sidecars.
+When using Istio `1.5+`, the `istiod` and `proxy` metricsets should be used. The `istiod` collects metrics directly from Istio Daemon Prometheus Exporter while `proxy` collects from the Prometheus Exporter of each of the proxy sidecars.
 
-The default metricsets are `mesh`, `mixer`, `pilot`, `galley`, `citadel`.
+The default metricsets are `mesh`, `mixer`, `pilot`, `galley`, `citadel`. If using Istio `1.6+`, you should specify the metricsets `istiod` and `proxy`.
```
Comment (ChrsMark) [inline metricbeat/docs/modules/istio.asciidoc:21]:
```suggestion
The default metricsets are `mesh`, `mixer`, `pilot`, `galley`, `citadel`. If using Istio `1.5+`, you should specify the metricsets `istiod` and `proxy`.
```

File: metricbeat/docs/modules/istio.asciidoc [hunk 2]
```diff
@@ -11,21 +11,20 @@ This file is generated! See scripts/mage/docs_collector.go
...
-module collects metrics from the
-Istio https://istio.io/v1.4/docs/tasks/observability/metrics/querying-metrics/#about-the-prometheus-add-on[prometheus exporters endpoints].
 
-For versions after `1.5`, `istiod` and `proxy` metricsets can be used.
+When using Istio `1.4-`, then the `mesh`, `mixer`, `pilot`, `galley`, `citadel` metricsets should be used. The Istio
+module collects metrics from these
+Istio https://istio.io/v1.4/docs/tasks/observability/metrics/querying-metrics/#about-the-prometheus-add-on[prometheus exporters endpoints].
 
-`istiod` collects metrics directly from Istio Daemon while `proxy` collects from each of the proxy sidecars.
+When using Istio `1.5+`, the `istiod` and `proxy` metricsets should be used. The `istiod` collects metrics directly from Istio Daemon Prometheus Exporter while `proxy` collects from the Prometheus Exporter of each of the proxy sidecars.
```
Comment (ChrsMark) [inline metricbeat/docs/modules/istio.asciidoc:19]: Maybe we can add that metrics can be found at https://istio.io/latest/docs/reference/config/metrics/.

**Review by ChrsMark — COMMENTED** [review]
Left one small correction. Also can we add a reference to https://istio.io/latest/docs/reference/config/metrics/ for the metrics that we collect with `istiod` and `proxy` metricsets?

File: metricbeat/docs/modules/istio.asciidoc (hunk 2 above)
Comment (cyrille-leclerc) [reply metricbeat/docs/modules/istio.asciidoc:19]: Please review sentence added below

**Comment by cyrille-leclerc** [comment]
@ChrsMark please review.

**Comment by ChrsMark** [comment]
@cyrille-leclerc  it looks good to me. However the linter is failing. Do you have the repo locally so as to run `make fmt update` to fix any linting issues? Otherwise I can cherry-pick your commits at https://github.com/elastic/beats/pull/29875 and ship them all together.

**Review by ChrsMark — CHANGES_REQUESTED** [review]
Linting issues should be fixed before merging.

**Comment by cyrille-leclerc** [comment]
Thanks @ChrsMark , after many efforts, I stil fail to run `make fmt update` with `mage fmt \n Unknown target specified: "fmt"`.

Can you please help me running this command on an environment that works? hopefully the command works on your environment.

details: https://gist.github.com/cyrille-leclerc/af93048975069a3191310cb150266ed6

**Comment by ChrsMark** [comment]
@cyrille-leclerc in which directory do you execute this command? It should be executed under `~/go/src/github.com/elastic/beats/metricbeat` (adjust accordingly to your home dir) and `~/go/src/github.com/elastic/beats/x-pack/metricbeat`.

**Comment by cyrille-leclerc** [comment]
@ChrsMark `make fmt update` on the `metricbeat`directory seems to revert my changes. Did I miss something?

```
diff --git a/metricbeat/docs/modules/istio.asciidoc b/metricbeat/docs/modules/istio.asciidoc
index 825af12127..aedb381098 100644
--- a/metricbeat/docs/modules/istio.asciidoc
+++ b/metricbeat/docs/modules/istio.asciidoc
@@ -11,21 +11,21 @@ This file is generated! See scripts/mage/docs_collector.go
 beta[]
 
 This is the Istio module.
-
-When using Istio `1.4-`, then the `mesh`, `mixer`, `pilot`, `galley`, `citadel` metricsets should be used. The Istio
-module collects metrics from these
+This module is compatible with versions before `1.5` of Istio where microservices architecture is used. If using
+versions prior to `1.5` then `mesh`, `mixer`, `pilot`, `galley`, `citadel` metricsets should be used where the Istio
+module collects metrics from the
 Istio https://istio.io/v1.4/docs/tasks/observability/metrics/querying-metrics/#about-the-prometheus-add-on[prometheus exporters endpoints].
 
-When using Istio `1.5+`, the `istiod` and `proxy` metricsets should be used. The `istiod` collects metrics directly from Istio Daemon Prometheus Exporter while `proxy` collects from the Prometheus Exporter of each of the proxy sidecars.
-The documentation of the collected metrics is available on https://istio.io/latest/docs/reference/config/metrics/[Istio Documentation > Reference > Configuration > Istio Standard Metrics].
+For versions after `1.5`, `istiod` and `proxy` metricsets can be used.
+
+`istiod` collects metrics directly from Istio Daemon while `proxy` collects from each of the proxy sidecars.
 
-The default metricsets are `mesh`, `mixer`, `pilot`, `galley`, `citadel`. If using Istio `1.5+`, you should specify the metricsets `istiod` and `proxy`.
+The default metricsets are `mesh`, `mixer`, `pilot`, `galley`, `citadel`.
 
 [float]
 === Compatibility
 
 The Istio module is tested with Istio `1.4` for `mesh`, `mixer`, `pilot`, `galley`, `citadel`.
-
 The Istio module is tested with Istio `1.7` for `istiod` and `proxy`.
 
 [float]
@@ -49,28 +49,7 @@ image::./images/metricbeat-istio-traffic.png[]
 === Example configuration
 
 The Istio module supports the standard configuration options that are described
-in <<configuration-metricbeat>>. 
-
-Example configuration for Istio `1.5+`:
-[source,yaml]
-----
-metricbeat.modules:
-# Istio istiod to monitor the Istio Daemon for versions after 1.5 of Istio.
-- module: istio
-  metricsets: ['istiod']
-  period: 10s
-  # use istiod.istio-system:15014, when deploying Metricbeat in a kubernetes cluster as Pod or Daemonset
-  hosts: ['localhost:15014']
-
-# Istio proxy to monitor Envoy sidecars for versions after 1.5 of Istio.
-- module: istio
-  metricsets: ['proxy']
-  period: 10s
-  # it's recommended to deploy this metricset with autodiscovery, see metricset's docs for more info
-  hosts: ['localhost:15090']
-----
-
-Example configuration for Istio `1.4-`:
+in <<configuration-metricbeat>>. Here is an example configuration:
 
 [source,yaml]
 ----
@@ -109,6 +88,20 @@ metricbeat.modules:
   period: 10s
   # use istio-pilot.istio-system:15014, when deploying Metricbeat in a kubernetes cluster as Pod or Daemonset
   hosts: ["localhost:15014"]
+
+# Istio istiod to monitor the Istio Daemon for versions after 1.5 of Istio.
+- module: istio
+  metricsets: ['istiod']
+  period: 10s
+  # use istiod.istio-system:15014, when deploying Metricbeat in a kubernetes cluster as Pod or Daemonset
+  hosts: ['localhost:15014']
+
+# Istio proxy to monitor Envoy sidecars for versions after 1.5 of Istio.
+- module: istio
+  metricsets: ['proxy']
+  period: 10s
+  # it's recommended to deploy this metricset with autodiscovery, see metricset's docs for more info
+  hosts: ['localhost:15090']
 ----
 
 This module supports TLS connections when using `ssl` config field, as described in <<configuration-ssl>>.
```

**Comment by ChrsMark** [comment]
@cyrille-leclerc just noticed that you are editing a wrong file. You need to apply the changes on https://github.com/elastic/beats/blob/38add00bffb8565c1b5e0cfe17776934af3ef525/x-pack/metricbeat/module/istio/_meta/docs.asciidoc (this is the source file let's say) and then run `make fmt update` so as to update the final file properly.

**Comment by cyrille-leclerc** [comment]
@ChrsMark please review updated docs

**Review by ChrsMark — APPROVED** [review]

**Pull request merged**